hls.js/index.d.ts

```diff
@@ -0,0 +1,523 @@
+            onProgress: (stats: LoaderStats, context: LoaderContext, data: string | ArrayBuffer) => void;
+            // Callback triggered if any I/O error is met while loading fragment.
+            onError: (err: { code: number, text: string }, context: LoaderContext ) => void;
+            // Callback triggered if loading is still not finished after a certain duration.
+            onTimeout: (stats: LoaderStats, context: LoaderContext) => void;
+        }): void;
+
+        /**
+         * Abort any loading in progress.
+         */
+        abort(): void;
+
+        /**
+         * Destroy loading context.
+         */
+        destroy(): void;
+    }
+
+    /**
+     * Config that gets passed into the HLS constructor. See the documentation
+     * for more details: https://github.com/dailymotion/hls.js/blob/master/API.md#fine-tuning
+     */
+    export interface Config {
+        autoStartLoad?: boolean;
+        startPosition ?: number;
+        capLevelToPlayerSize?: boolean;
+        debug?: boolean;
+        defaultAudioCodec?: string;
+        maxBufferLength?: number;
+        maxMaxBufferLength?: number;
+        maxBufferSize?: number;
+        maxBufferHole?: number;
+        maxSeekHole?: number;
+        seekHoleNudgeDuration?: number;
+        nudgeOffset?: number;
+        nudgeMaxRetry?: number;
+        maxFragLookUpTolerance?: number;
+        liveSyncDurationCount?: number;
+        liveMaxLatencyDurationCount?: number;
+        enableWorker?: boolean;
+        enableSoftwareAES?: boolean;
+        manifestLoadingTimeOut?: number;
+        manifestLoadingMaxRetry?: number;
+        manifestLoadingRetryDelay?: number;
+        manifestLoadingMaxRetryTimeout ?: number;
+        levelLoadingTimeOut?: number;
+        levelLoadingMaxRetry?: number;
+        levelLoadingRetryDelay?: number;
+        levelLoadingMaxRetryTimeout?: number;
+        fragLoadingTimeOut?: number;
+        fragLoadingMaxRetry?: number;
+        fragLoadingRetryDelay?: number;
+        fragLoadingMaxRetryTimeout?: number;
+        fragLoadingLoopThreshold?: number;
+        startFragPrefech?: boolean;
+        appendErrorMaxRetry?: number;
+        loader?: Loader;
+        fLoader?: Loader;
+        pLoader?: Loader;
+        xhrSetup?: (xhr?: XMLHttpRequest, url?: string) => void;
+        fetchSetup?: (context?: LoaderContext, params?: any) => void;
+        abrController?: any;
+        timelineController?: any;
+        enableCEA708Captions?: boolean;
+        stretchShortVideoTrack?: boolean;
+        forceKeyFrameOnDiscontinuity?: boolean;
+        abrEwmaFastLive?: number;
+        abrEwmaSlowLive?: number;
+        abrEwmaFastVoD?: number;
+        abrEwmaSlowVoD?: number;
+        abrEwmaDefaultEstimate?: number;
+        abrBandWidthFactor?: number;
+        abrBandWidthUpFactor?: number;
+    }
+
+    export interface ErrorEvent {
+        type: ErrorType;
+        details: ErrorDetails;
+        fatal: boolean;
+    }
```

Comment:
Some LoadError has `url` and `response`, `loader` property.

For example, `Hls.ErrorDetails.MANIFEST_LOAD_ERROR`  has `url` and `response` object.

```
Hls.ErrorDetails.MANIFEST_LOAD_ERROR - raised when manifest loading fails because of a network error

    data: { type : NETWORK_ERROR, details : Hls.ErrorDetails.MANIFEST_LOAD_ERROR, fatal : true, url : manifest URL, response : { code: error code, text: error text }, loader : URL loader }
```

- https://github.com/dailymotion/hls.js/blob/master/doc/API.md#network-errors

Probably, `Hls.ErrorDetails.FRAG_PARSING_ERROR` and `Hls.ErrorDetails.FRAG_LOAD_ERROR` has a specific property.